File: tm/NativeSampleModule.ts
import {TurboModule, TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule {
  readonly reverseString: (input: string) => string;
  readonly addTwoNumbers: (a: number, b: number) => number;
}

export default TurboModuleRegistry.getEnforcing<Spec>('NativeSampleModule');

File: tm/NativeSampleModule.cpp
#include "NativeSampleModule.h"

namespace facebook::react {

NativeSampleModule::NativeSampleModule(std::shared_ptr<CallInvoker> jsInvoker)
    : NativeSampleModuleCxxSpec(std::move(jsInvoker)) {}

std::string NativeSampleModule::reverseString(jsi::Runtime& rt, std::string input) {
  return std::string(input.rbegin(), input.rend());
}

// Add this inside the NativeSampleModule namespace in NativeSampleModule.cpp

int NativeSampleModule::addTwoNumbers(jsi::Runtime& rt, int a, int b) {
    return a + b;
}


} // namespace facebook::react